doc/generated_testcases_v2.yaml

```diff
@@ -0,0 +1,221 @@
+# The top-level is always assumed to be a directory.
+      hard_cases_group:
+# TODO: We could drop this because testcase dictionaries need the input: key
+# while testgroups need the include: and/or data: keys.
+        type: directory
+
+# Directories may contain a testdata.yaml that will be written as specified.
+        testdata.yaml:
+          on_reject: break
+          accept_score: 25
+          range: 0 25
+          grader_flags: min
+
+# A testgroup may include all cases in another testgroup. All files for each
+# testcase in the other testgroup will be symlinked into the current testgroup.
+# Nested included testcases are also included here.
+#
+# This must be a list of testgroups and/or testcases paths relative to data/.
+# Testcases must be referred to by their {name}.
+#
+# You may only include testgroups that come before the current group, either
+# earlier in a list or earlier alphabetically in dictionaries.
+#
+# TODO: How does this interact with numbering? How do we guarantee included
+# cases come first?
+# include numbered into numbered: OK
+# include dict into dict:     error on name collision. How to guarantee they
+# come first?
+# include numbered into dict: error on name collision. Do we allow this?
+# include dict into numbered: error on name collision. Do we allow this?
+        include:
+          - sample
+          - secret/03
+        
+# To enable automatic numbering of testcases, data: may also contain a list of
+# dictionaries instead of a single dictionary. In this case, testcases and/or
+# groups will be numbered in the order they appear. The system will determine
+# the required number of digits to use and numbers will be zero-padded
+# accordingly, using a dash as separator from the given name. All items in a
+# given dictionary will get the same number. Use a list of 1-item dictionaries
+# for incremental numbering.
+#
+# Any testgroup inside a list must contain a list object as data:.
+#
+# Numbering is global. Testcases/testgroups are ordered by the order of lists
+# and alphabetical for dictionaries. Numbering is done in pre-order: first
+# number the current testgroup, then number everything inside the group, then
+# number testcases/testgroups following the current group.
+        data:
+          - a: stdout.py a
+            b: stdout.py b
+          - testgroup:
+              type: directory
+              data:
+                - c: stdout.py c
+# This is forbidden because data: dictionaries may not appear within data: lists.
+             #data:
+             #  c: stdout.py c
+          - d: stdout.py d
+
+# The above data: list is equivalent to the map:
+       #data:
+       #  1-a: stdout.py a
+       #  1-b: stdout.py b
+       #  2-testgroup:
+       #    type: directory
+       #    data:
+       #      3-c: stdout.py c
+       #  4-d: stdout.py d
+       
+
+# At the directory level, unknown keys are allowed for tooling-specific
+# extensions.
```

Comment:
I guess we'd also want to allow it at the global level, just like already existing keys like `solution` and `visualizer`.